src/odbc_connection.cpp

```diff
@@ -674,14 +471,156 @@ Napi::Value ODBCConnection::CreateStatement(const Napi::CallbackInfo& info) {
+  SQLULEN      fetch_size;
+} QueryOptions;
+
+#define QUERY_OPTION_PROPERTY_NAME_CURSOR     "cursor"
+#define QUERY_OPTION_PROPERTY_NAME_FETCH_SIZE "fetchSize"
+
+QueryOptions
+parse_query_options
+(
+  Napi::Env   env,
+  Napi::Value options_value
+)
+{
+  QueryOptions query_options;
+
+  if (options_value.IsNull())
+  {
+    query_options.use_cursor  = false;
+    query_options.cursor_name = NULL;
+
+    return query_options;
+  }
+
+  Napi::Object options_object = options_value.As<Napi::Object>();
+
+  // .cursor property
+  if (options_object.HasOwnProperty(QUERY_OPTION_PROPERTY_NAME_CURSOR))
+  {
+    Napi::Value cursor_value =
+      options_object.Get(QUERY_OPTION_PROPERTY_NAME_CURSOR);
+    
+    if (cursor_value.IsString())
+    {
+      query_options.use_cursor  = true;
+#ifdef UNICODE
+      std::u16string cursor_string;
+      cursor_string = cursor_value.As<Napi::String>().Utf16Value();
+#else
+      std::string cursor_string;
+      cursor_string = cursor_value.As<Napi::String>().Utf8Value();
+#endif
+      query_options.cursor_name = (SQLTCHAR *)cursor_string.c_str();
+      query_options.cursor_name_length = cursor_string.length();
+    }
+    else if (cursor_value.IsBoolean())
+    {
+      query_options.use_cursor  = cursor_value.As<Napi::Boolean>().Value();
+      query_options.cursor_name = NULL;
+    }
+    else
+    {
+      Napi::TypeError::New(env, "Connection.query options: ." QUERY_OPTION_PROPERTY_NAME_CURSOR " must be a STRING or a BOOLEAN value.").ThrowAsJavaScriptException();
+    }
+  }
+  else
+  {
+    query_options.use_cursor  = false;
+    query_options.cursor_name = NULL;
+  }
+  // END .cursor property
+
+  // .fetchSize property
+  if (options_object.HasOwnProperty(QUERY_OPTION_PROPERTY_NAME_FETCH_SIZE))
+  {
+    Napi::Value fetch_size_value =
+      options_object.Get(QUERY_OPTION_PROPERTY_NAME_FETCH_SIZE);
+
+    if (fetch_size_value.IsNumber())
+    {
+      // even if the user didn't explicitly set use_cursor to true, if they are
+      // passing a fetch size, it should be assumed.
+      query_options.use_cursor = true;
+
+      query_options.fetch_size =
+        (SQLULEN)fetch_size_value.As<Napi::Number>().Int64Value();
+
+      // Having a fetch size less than 1 doesn't make sense, so switch to a sane
+      // value.
+      if (query_options.fetch_size < 1)
+      {
+        query_options.fetch_size = 1;
+      }
+    }
+    else
+    {
+      Napi::TypeError::New(env, "Connection.query options: ." QUERY_OPTION_PROPERTY_NAME_FETCH_SIZE " must be a NUMBER value.").ThrowAsJavaScriptException();
+      return query_options;
+    }
+  }
+  else
+  {
+    query_options.fetch_size = 1;
+  }
+  // END .fetchSize property
+
+
+  return query_options;
+}
+
+SQLRETURN
+set_fetch_size
+(
+  StatementData *data,
+  SQLULEN        fetch_size
+)
+{
+  data->sqlReturnCode =
+  SQLSetStmtAttr
+  (  
+    data->hSTMT,  
+    SQL_ATTR_ROW_ARRAY_SIZE,  
+    (SQLPOINTER) fetch_size,  
+    0
+  );
```

Comment:
FYI, probably should check for warning 01S02 - Option value changed and call SQLGetStmtAttr to get the adjusted fetch size.